### PR TITLE
frontends: dark mode

### PR DESCRIPTION
**Pull request by jadzeidan**
This commit contains refactoring required for dark mode and changes in the style sheet so the variables use dark mode colors.

There are still more parts that need to be done, such as:
- change the chart background color to a dark mode color
- change color of SVGs (such as "x" in dialogs and "save" button in notes)
- make dark mode images (such as logos)

### CODE REVIEW

**Comment by jadzeidan** [comment]
For SVGs, we currently use `` tags. Does it make sense to convert certain ones (such as the "x" in dialogs") to `<svg>` tags instead? Then we can use CSS to color them.

File: frontends/web/src/components/icon/icon.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ import closeXWhiteSVG from './assets/icons/close-x-white.svg';
 import closeXDarkSVG from './assets/icons/close-x-dark.svg';
 import guideSVG from './assets/icons/guide.svg';
 import menuSVG from './assets/icons/menu.svg';
```
Comment (shonsirsha) [inline frontends/web/src/components/icon/icon.tsx:35]: we could remove this unused `menuSVG` import.

Comment (shonsirsha) [inline frontends/web/src/components/icon/icon.tsx:35]: we could remove this unused `menuSVG` import.

**Review by shonsirsha — CHANGES_REQUESTED** [review]
Nice, good job so far! 👍 

prob still in WIP, regardless left a small comment.

File: frontends/web/src/components/icon/icon.tsx (hunk 1 above)
Comment (jadzeidan) [reply frontends/web/src/components/icon/icon.tsx]: removed now

File: frontends/web/src/routes/account/summary/chart.module.css [hunk 2]
```diff
@@ -142,8 +142,8 @@
   
 
 .tooltip {
-  background: white;
-  border: 1px solid #dedede;
+  background: var(--background-secondary);
+  border: 1px solid var(--background)
```
Comment (shonsirsha) [inline frontends/web/src/routes/account/summary/chart.module.css:146]: missed a `;` here 😉

**Review by shonsirsha — CHANGES_REQUESTED** [review]
minor one

File: frontends/web/src/routes/account/summary/chart.module.css (hunk 2 above)
Comment (jadzeidan) [reply frontends/web/src/routes/account/summary/chart.module.css]: Fixed

**Comment by shonsirsha** [comment]
BTW looks good. I'll try to have a look again today on the code. Can you change this PR to be merging into staging-darkmode instead of master though? Thx!

**Comment by shonsirsha** [comment]
We can safely close this due to: #1912 🙂

**Pull request closed without merging**